### PR TITLE
Changed ReadLsDyna (for coordinates array initializing)

### PR DESCRIPTION
**Pull request by askarali82**
Please test with large .k files.

### CODE REVIEW

File: src/io/input/ReadLsDyna.cpp [hunk 1]
```diff
@@ -4,6 +4,13 @@
 #define TITLE_NODE     "*NODE"
 #define TITLE_END      "*END"
 
+//-------------------------------------------------------------------------------------------
+static int  CmpFunc(const void *item1, const void *item2) {
+    const double **val1 = (const double **)item1;
+    const double **val2 = (const double **)item2;
```
[the remark this answers is not in the document]
Comment (enstrophy) [reply src/io/input/ReadLsDyna.cpp]: @askarali82 : The PartitionMesh.cpp has the same compare function but with single pointer.
We can also think of moving such functions with global use into a separate header and have an accompanying cpp. @rhk12

File: src/io/input/ReadLsDyna.cpp [hunk 2]
```diff
@@ -144,58 +184,42 @@ bool ReadLsDyna(const char *FileName) {
...
-            if (strstr(Line, "z") != NULL) {
-                ndim = ndim + 1;
-            }
-            if (ndim == 2 || ndim == 3) {
-                coordinates = (double *)calloc(ConnectivitySize * ndim, sizeof(double));
-            }
-            continue;
+    i = 0;
+    Ended = false;
+    double **Nodes = (double **)malloc(AllNodesCount * sizeof(double *));
```
[the remark this answers is not in the document]
Comment (askarali82) [reply src/io/input/ReadLsDyna.cpp]: OK. I removed.

File: src/io/input/ReadLsDyna.cpp [hunk 3]
```diff
@@ -4,6 +4,13 @@
 #define TITLE_NODE     "*NODE"
 #define TITLE_END      "*END"
 
+//-------------------------------------------------------------------------------------------
+static int  CmpFunc(const void *item1, const void *item2) {
+    const double **val1 = (const double **)item1;
```
[the remark this answers is not in the document]
Comment (askarali82) [reply src/io/input/ReadLsDyna.cpp]: OK. Removed.

File: src/io/input/ReadLsDyna.cpp [hunk 4]
```diff
@@ -144,58 +184,42 @@ bool ReadLsDyna(const char *FileName) {
...
-            break;
+    }
+    qsort(Nodes, AllNodesCount, sizeof(double *), CmpFunc);
+    nnodes = 0;
+    const int csize = ndim * sizeof(double);
+    coordinates = (double *)malloc(ConnectivitySize * csize);
+    for (int i = 0; i < ConnectivitySize; i++) {
+        const double Key = connectivity[i] + 1;
+        const double *PKey = &Key;
+        const double **bs = (const double **)bsearch(&PKey, Nodes, AllNodesCount, sizeof(double *), CmpFunc);
```
[the remark this answers is not in the document]
Comment (askarali82) [reply src/io/input/ReadLsDyna.cpp]: Removed.

File: src/io/input/ReadLsDyna.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (askarali82) [reply src/io/input/ReadLsDyna.cpp]: Removed.

**Comment by askarali82** [comment]
I made the changes.

**Pull request merged**